### PR TITLE
Allow items to not have sourceIdentifiers

### PR DESCRIPTION
**Pull request by alicefuzier**
For #2474 

🎸 🕹

### CODE REVIEW

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/MiroTransformableTransformer.scala [hunk 1]
```diff
@@ -69,6 +69,7 @@ class MiroTransformableTransformer
           language = None,
           dimensions = None,
           items = getItems(miroData, miroTransformable.sourceId),
+          itemsV1 = getItemsV1(miroData, miroTransformable.sourceId),
```
Comment (alexwlchan) [inline catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/MiroTransformableTransformer.scala]: keyword arguments please? :)

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/miro/MiroImageApiURL.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    val iiifImageApiBaseUri = "https://iiif.wellcomecollection.org"
+
+    val imageUriTemplates = Map(
+      "thumbnail" -> "%s/image/%s.jpg/full/300,/0/default.jpg",
+      "info" -> "%s/image/%s.jpg/info.json"
+    )
+
+    val imageUriTemplate = imageUriTemplates.getOrElse(
+      templateName,
+      throw new Exception(
```
Comment (alexwlchan) [inline catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/miro/MiroImageApiURL.scala]: NoSuchElementException would be more specific, and a bit nicer.

File: sbt_common/internal_model/src/test/scala/uk/ac/wellcome/models/work/internal/WorkTest.scala [hunk 3]
```diff
@@ -59,6 +59,7 @@ class WorkTest extends FunSpec with Matchers with IdentifiersUtil {
     val caught = intercept[JsonDecodingError] {
       fromJson[IdentifiedWork](jsonString).get
     }
-    caught.getMessage shouldBe "Attempt to decode value on failed cursor: DownField(agent),DownArray,DownField(items)"
+    caught.getMessage should startWith(
+      "Attempt to decode value on failed cursor")
```
Comment (alexwlchan) [inline sbt_common/internal_model/src/test/scala/uk/ac/wellcome/models/work/internal/WorkTest.scala]: nit: Can you pull this onto a single line?

Comment (alicefuzier) [reply sbt_common/internal_model/src/test/scala/uk/ac/wellcome/models/work/internal/WorkTest.scala]: It was. The scala formatter broke it. IMHO the line length limit should be higher...

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/MiroTransformableTransformer.scala (hunk 1 above)
Comment (alicefuzier) [reply catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/MiroTransformableTransformer.scala]: I don't think they add anything here, to be honest, other than making the line longer

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/miro/MiroImageApiURL.scala (hunk 2 above)
Comment (alicefuzier) [reply catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/miro/MiroImageApiURL.scala]: Agree, but that's just a copy paste of the current logic somewhere else. I actually wonder what happens in that case...does it go in the DLQ? it should have a test for it probably..

**Pull request merged**